new-dti-website/src/app/layout.tsx

```diff
@@ -1,22 +1,58 @@
+'use client';
+
 import './globals.css';
 import { Inter, IBM_Plex_Mono } from 'next/font/google';
+import { usePathname } from 'next/navigation';
+import { useEffect } from 'react';
 import Page from '../../components/page';
 
+declare global {
+  interface Window {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    gtag: any;
+  }
+}
+const MEASUREMENT_ID = 'G-6Y507Z87ST';
 const inter = Inter({ subsets: ['latin'] });
 
 export const ibm_plex_mono = IBM_Plex_Mono({
   subsets: ['latin'],
   weight: ['400']
 });
 
-const RootLayout = ({ children }: { children: React.ReactNode }): JSX.Element => (
-  <html lang="en">
-    <body className={`${inter.className} bg-black`}>
-      <div className="relative overflow-x-hidden">
-        <Page>{children}</Page>
-      </div>
-    </body>
-  </html>
-);
+const RootLayout = ({ children }: { children: React.ReactNode }): JSX.Element => {
+  const pathname = usePathname();
+
+  const handleRouteChange = (path: string) => {
+    window.gtag('config', MEASUREMENT_ID, {
+      page_path: path
+    });
+  };
+
+  useEffect(() => {
```

Comment:
This looks great to me Andrew! I was just wondering if were possible to make this into a hook maybe like 
` useGoogleAnalytics();` with `handleRouteChange` so the layout is more focused on rendering the components? I think this is definitely a future thing though! 

```suggestion
  useGoogleAnalytics(MEASUREMENT_ID);
```